arch/arm/src/nrf52/nrf52_tim_lowerhalf.c

```diff
@@ -361,13 +361,13 @@ static int nrf52_timer_settimeout(struct timer_lowerhalf_s *lower,
       goto errout;
     }
 
-  if (timeout > NRF52_TIMER_MAX)
+  cc = (timeout * NRF52_TIMER_PER / 1000000);
```

Comment:
Shouldn't this be `if (timeout > NRF52_TIMER_MAX * 1000000 / NRF52_TIMER_PER)`? And calculate `cc` only in case of `false`?

Reply:
That's right, fixed. In this case there is no point in using 64 bit for `cc` variable